### PR TITLE
added a test for blocks over abstract class accepting itself as a clo…

### PR DESCRIPTION
**Pull request by uujava**
Just an additional test for blocks. It fails in previous versions of mirah, but works now.

### CODE REVIEW

**Comment by uujava** [comment]
have to compile java8 fixtures separatly. let me fix this.

**Comment by uujava** [comment]
added support for Java8 fixtures compilation

File: Rakefile [hunk 1]
```diff
@@ -120,14 +120,22 @@ end
 
 
 task :build_test_fixtures => 'tmp_test/fixtures/fixtures_built.txt'
-directory 'tmp_test/fixtures'
-file 'tmp_test/fixtures/fixtures_built.txt' => ['tmp_test/fixtures'] + Dir['test/fixtures/**/*.java'] do
+  directory 'tmp_test/fixtures'
```
Comment (baroquebobcat) [inline Rakefile]: nit: rm unnecessary indent.

**Comment by baroquebobcat** [comment]
Looks pretty good. Could you squash your commits together so there's just one?

**Pull request closed without merging**